### ISSUE TITLE
Subdomain registrar improvement

### ISSUE THREAD

**Issue by Akirtovskis** [stacksgov/Stacks-Grant-Launchpad#282]
**Background**

Increasing the performance of the subdomain registrar to improve user onboarding on Sigle and on other apps which use the subdomain registrar https://github.com/stacks-network/subdomain-registrar.
The performance of the subdomain is currently an issue, with the API taking more than 2mins to send a response some days. The stability is also a concern, from our tests, 1 out of 5 request is failing.

**Project Overview**<br/>
What solution are you providing? Who will it serve?

Cleanup and increase performance for the subdomain registrar so new projects can onboard users much faster.

**Scope**<br/>
What are the components or technical specs of the project? What will the final deliverable look like? How will you measure success?

Final deliverable will be commited to the main repo.

- Improved performance of registration and stability
- Code migrated to TypeScript
- Database implementation migrated to Postgres

**Budget and Milestones**<br/>
What grant amount are you seeking? How long will the project take in hours? If more than 20, please break down the project into milestones, with a clear output (e.g., low-fi mockup, MVP with two features) and include the estimated work hours for each milestone.

Total Grant Request: $5k

M1: Migrate types from Flow to TypeScript and improve tooling in the repo (prettier, eslint) (3 days)

M2: Migrate usage from the `blockstack` SDK to the new stacks.js (3 days)

M3: Database migration from sqlLite to Postgres (5 days)

Each milestone can be done as separate pull request to the Stacks repo and would require review / approval from the Hiro team before the next milestone start.

**Team**<br/>
Who is building this? What relevant experience do you bring to this project? Are there skills sets you are missing that you are seeking from the community? Please share links to previous work.

Will be executed by Sigle dev team that has experience both working with the technologies which will be implemented and also Stacks repositories.

- https://www.sigle.io/
- https://www.stxstats.co/

**Risks**<br/>
What dependencies or obstacles do you anticipate? What contingency plans do you have in place?

Main risks will be communication and time for review coming from the Hiro team. This could move a lot the deadline of this task.

**Community and Supporting Materials**<br/>
Do you have previous projects, code commits, or experiences that are relevant to this application? What community feedback or input have you received? How do you plan to share your plan to the community over time and as the final deliverable?

You can see some examples of projects we did here [Sigle](https://github.com/sigle).
Before starting any work on this task we need to validate the plan with Hiro engineers and agree with the implementation approach for postgres.

**Comment by jruffer** [comment]
Is this what btc.us using?  I notice that the documentation is not up to date and I would like to learn more about this project. Is the community allowed to fund this?

**Comment by 314159265359879** [comment]
> Is this what btc.us using? I notice that the documentation is not up to date and I would like to learn more about this project. Is the community allowed to fund this?

btc.us uses BNS names. BNS names can have subdomains too however. For example if you own id.btc a registar could enable you or you could facilitate other users to make subdomains on that BNS name such as jruffer.id.btc (pointing jruffer.id.btc to a public key)

Pravica has a registar working for id.stx for example in their mobile dapp. They also had a grant for that by the way as you can see here: https://github.com/stacksgov/Stacks-Grants/issues/134 I assume you knew about this @Akirtovskis.

**Comment by will-at-stacks** [comment]
Hi @Akirtovskis ,

Thanks for your grant submission! We understand any extra ms shaved off can greatly improve user experience. We are happy to approve this grant and look forward to the performance and stability improvements this will bring!

Please let us know if you feel further development on this front is something you want to take on and we can work to support it.
Best,
Will

**Comment by Akirtovskis** [comment]
@will-at-stacks hey! 

Thanks for approving the grant! We are excited to proceed further!

However, I just tried to log in with Github using the link above and it didn't work. 

This is what came up:
```
No Pending Grants
We’re sorry, no onboarding is ready for you.
Please comment on the Github issue
if experiencing issues.
```

**Comment by Akirtovskis** [comment]
Fixed now, just filled in the form.

**Comment by Akirtovskis** [comment]
Contract signed and already kicked off discussion with the relevant devs from Stacks to be aligned with them.

https://github.com/stacks-network/subdomain-registrar/issues/77

Will start on M1 once they give us green light to do so. All the subsequent milestones depend on this one, so we are planning accordingly.

**Comment by will-at-stacks** [comment]
Hi @Akirtovskis thanks for tagging the grant issue in the related repo.  Since I'm not too fluent in reading code (as in, I can't at all lol) could you give me a quick update on the grants progress?  Thanks, Will

**Comment by Akirtovskis** [comment]
Hey @will-at-stacks sure thing!

Current state of M1 is that it's still in progress with one PR being absolutely ready for review. 

This is the one and it adds prettier + eslint config which is part of the dev experience and performance
improvements mentioned above:
https://github.com/stacks-network/subdomain-registrar/pull/80

This is the second PR which is still in WIP state, but soon will be done. It adds the Typescript config 
initial setup and once reviewed, will improve every devs experience there as well:
https://github.com/stacks-network/subdomain-registrar/pull/82

Will keep you posted once there is more to report.

**Comment by will-at-stacks** [comment]
Hello and thank you for participating in the Stacks Foundation Grants Program!

We are in the process of migrating from GitHub to the new [Grants Dashboard](https://grants.stacks.org/). In order to complete your grant, you will need to submit any remaining Progress Review and/or Final Review requests through the Dashboard in order to receive your remaining payments.

- If you have questions about the Grants Dashboard in general please email [ambassadors@stacks.org](mailto:ambassadors@stacks.org).
- If you have questions regarding payments and/or contracts please email [payments@stacks.org](mailto:payments@stacks.org).

Lastly, please note we are marking this grant 'closed' on GitHub for organizational purposes, but it is still 'open' on the Grants Dashboard.

Thanks and we hope to continue to support your efforts with additional grants!

Best,
Will
